corehq/motech/value_source.py

```diff
@@ -76,7 +76,7 @@ class ValueSource(DocumentSchema):
 
     def __eq__(self, other):
         return (
-            isinstance(other, ValueSource)
+            isinstance(other, self.__class__)
```

Comment:
nit: Someone recently [pointed out to me in an SO comment](https://stackoverflow.com/questions/2909106/whats-a-correct-and-good-way-to-implement-hash/19073010?noredirect=1#comment100385743_19073010) that the convention is to return `NotImplemented` rather than comparing types in `__eq__`.
```py
if not isinstance(other, self.__class__):
    return NotImplemented
return (
    self.doc_type == other.doc_type
    and ...
)
```
This allows Python to fall back to other `__eq__` implementations on superclasses. It's probably not critical here.

Reply:
Thanks. That's really useful to know. ... e7d0d9d0baf